include/spdlog/details/file_helper-inl.h

```diff
@@ -54,6 +54,11 @@ SPDLOG_INLINE void file_helper::open(const filename_t &fname, bool truncate)
             }
             std::fclose(tmp);
         }
+        
+        if (event_handlers_.before_open)
```

Comment:
I think its better called before the loop, otherwise it would be calked multiple times

Reply:
god, my mistake~~

fixed.